programs/cutting_boards.py

```diff
@@ -0,0 +1,53 @@
+""""
+Alice gives Bob a board composed of  1 x 1 wooden squares and asks him to find the minimum cost of breaking the board back down 
+into its individual squares. To break the board down, Bob must make cuts along its horizontal and vertical lines.
+
+To reduce the board to squares, Bob makes horizontal and vertical cuts across the entire board. Each cut has a given cost,
+or for each cut along a row or column across one board, so the cost of a cut must be multiplied by the number of segments 
+it crosses. The cost of cutting the whole board down into  squares is the sum of the costs of each successive cut.
+
+example
+
+6 4  (dimentions)
+we will receive a cost_y and cost_x lists than indicated the cost of each cut and how many cuts we need
+
+cost_y = [2 1 3 1 4]
+cost_x = [4 1 2]
+
+"""
+
+def boardCutting(cost_y: list, cost_x: list)-> int:
+    # Write your code here
+    cost = 0
+    cut_y = 0
+    cut_x = 0
+    cost_y.sort(reverse=True)
+    cost_x.sort(reverse=True)
+    while cost_y and cost_x:
+        if cost_y[0] > cost_x[0] or (cost_y[0] == cost_x[0] and cut_x < cut_y):
+            c = cost_y.pop(0)
```

Comment:
Don't use `pop(0)` EVER

This has O(n) complexity every time. 

For this problem, don't sort in reverse if you need to pop, so you can pop the last